new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSessionMigration.java

```diff
@@ -0,0 +1,460 @@
+  private static ConditionFactory withStandardRetryPolicy = null;
+
+  /**
+   * Install Operator, create a custom image using model in image with model files
+   * and create a one cluster domain.
+   *
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+   *                   JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void init(@Namespaces(2) List<String> namespaces) {
+    // create standard, reusable retry/backoff policy
+    withStandardRetryPolicy = with().pollDelay(2, SECONDS)
+      .and().with().pollInterval(10, SECONDS)
+      .atMost(5, MINUTES).await();
+
+    // get a unique Operator namespace
+    logger.info("Get a unique namespace for Operator");
+    assertNotNull(namespaces.get(0), "Namespace list is null");
+    opNamespace = namespaces.get(0);
+
+    // get a unique domain namespace
+    logger.info("Get a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domainNamespace = namespaces.get(1);
+
+    // install and verify Operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // create and verify WebLogic domain image using model in image with model files
+    String imageName = createAndVerifyDomainImage();
+
+    // create and verify one cluster domain
+    logger.info("Create domain and verify that it's running");
+    createAndVerifyDomain(imageName);
+
+    // map to save HTTP response data
+    httpAttrMap = new HashMap<String, String>();
+    httpAttrMap.put("sessioncreatetime", "(.*)sessioncreatetime>(.*)</sessioncreatetime(.*)");
+    httpAttrMap.put("sessionid", "(.*)sessionid>(.*)</sessionid(.*)");
+    httpAttrMap.put("primary", "(.*)primary>(.*)</primary(.*)");
+    httpAttrMap.put("secondary", "(.*)secondary>(.*)</secondary(.*)");
+    httpAttrMap.put("count", "(.*)countattribute>(.*)</countattribute(.*)");
+  }
+
+  @AfterAll
+  void tearDown() {
+  }
+
+  /**
+   * Get the primary and secondary server name and session create time from the factory method,
+   * getOriginalServerAndSessionInfo. Stop the primary server by changing ServerStartPolicy to NEVER
+   * and patching domain. Send a HTTP request to get http session state (count number), primary server
+   * and session create time. Verify that a new primary server is picked and HTTP session state is migrated.
+   *
+   * @param origPrimaryServerName the primary server name from the factory method
+   *                              getOriginalServerAndSessionInfo when the domain is created
+   * @param origSecondaryServerName the secondary server name from the factory method
+   *                                getOriginalServerAndSessionInfo when the domain is created
+   * @param origPrimaryServerName the session create time from the factory method
+   *                              getOriginalServerAndSessionInfo when the domain is created
+   */
+  @ParameterizedTest
```

Comment:
why is this a parameterized test? what are the different parameters we are testing with?

Reply:
changed